### PR TITLE
improve UI/UX around new/edit admin templates

### LINKED ISSUE #963: When editing a template save/cancel hidden unless change is made
### Current Behavior

When trying to edit a template there is no obvious way to exit if no changes are made.

You can use the sidebar menu, but it feels strange to not be able to close the edit screen.

### Expected Behavior

Cancel & Save should always be shown, but save should be disabled until there is a change

### Steps To Reproduce

As admin
View Templates
Edit an existing template
Make no changes.

### Environment

- FlowForge version: 0.9
- Node.js version: 16.x
- npm version: 8.x
- Platform/OS: Linux
- Browser: Chrome

### PR DESCRIPTION
**Pull request by Steve-Mcl**
fixes #963

@hardillb  I discussed this with @joepavitt earlier today & we agreed several improvements 

1. Change button name of "Edit Template" to "Discard Changes"
2. Always show the "Save Changes" button, but conditionally enabled/disabled if fields have changed
3. Hid the "Discard Changes" button, and only show it if fields have changed
4. Add a kinda mini breadcrumb that takes you back to templates list
5. Make items in templates list clickable to default operation (edit)  (saves 1 click & moving mouse to far side)

In action...

Image

NOTE: Needs https://github.com/flowforge/forge-ui-components/issues/58 to be fixed before merging

### CODE REVIEW

File: frontend/src/pages/admin/Template/index.vue [hunk 1]
```diff
@@ -276,12 +290,18 @@ export default {
                     name: 'Admin Templates'
                 })
             } catch (err) {
+                if (err.response?.data) {
+                    alerts.emit(err.response.data.error, 'warning')
+                } else {
+                    alerts.emit('Unknown Error. Check logs.', 'warning')
```
Comment (Pezmc) [inline frontend/src/pages/admin/Template/index.vue:296]: This is a potentially user-facing error, right? As far as I understand they wouldn't be able to see the logs. Is there other information we could surface to help them (and us) when they ultimately write into support if it keeps happening?

File: frontend/src/pages/admin/Templates/index.vue [hunk 2]
```diff
@@ -13,7 +14,9 @@
         <ff-loading v-if="loading" message="Loading Templates..." />
         <ff-data-table v-if="!loading" :columns="columns" data-el="templates"
                        :rows="templates" :show-search="true" search-placeholder="Search Templates..."
-                       :search-fields="['name',, 'description', 'owner_username', 'owner_id']">
+                       :search-fields="['name',, 'description', 'owner_username', 'owner_id']"
```
Comment (Pezmc) [inline frontend/src/pages/admin/Templates/index.vue:17]: Is the double comma a typo here? I'm not aware of some sort of special vue syntax.

Comment (Steve-Mcl) [reply frontend/src/pages/admin/Templates/index.vue:17]: unknown. existing code. suspect it is but it has zero affect.

Comment (joepavitt) [reply frontend/src/pages/admin/Templates/index.vue:17]: yep, legacy typo

Comment (joepavitt) [inline frontend/src/pages/admin/Templates/index.vue:17]:
```suggestion
                       :search-fields="['name', 'description', 'owner_username', 'owner_id']"
```

**Review by joepavitt — CHANGES_REQUESTED** [review]

File: frontend/src/pages/admin/Template/index.vue (hunk 1 above)
Comment (Pezmc) [reply frontend/src/pages/admin/Template/index.vue:296]: Let's leave it as it is, other than adding some extra fluff words ("An unhandled error occurred, please try again or check the logs."), I can't think of a better suggestion right now!

**Review by joepavitt — APPROVED** [review]

**Comment by joepavitt** [comment]
Approved - although now noticed the E2E is failing - holding off from merge

**Review by joepavitt — CHANGES_REQUESTED** [review]
Need to fix https://github.com/flowforge/forge-ui-components/issues/58 in order to rescue the E2E tests, or backtrack on the `rows-selectable` feature introduced here until that is fixed.

File: frontend/src/pages/admin/Template/index.vue (hunk 1 above)
Comment (Steve-Mcl) [reply frontend/src/pages/admin/Template/index.vue:296]: Another thing I forgot to mention... Once the validation is fixed on this form (see #1067) neither of the toast notifications will be seen unless there are exceptional circumstances.

**Review by Pezmc — APPROVED** [review]
https://github.com/flowforge/forge-ui-components/issues/58 is now fixed and I've pulled the latest version into this PR

**Review by joepavitt — APPROVED** [review]

**Pull request merged**